src/ReturnBookDao.java

```diff
@@ -0,0 +1,50 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+public class ReturnBookDao {
+	public static int delete(String bookcallno,int studentid){
+		int status=0;
+		try{
+			Connection con=DB.getConnection();
+			
+			status=updatebook(bookcallno);//updating quantity and issue
+			
+			if(status>0){
+			PreparedStatement ps=con.prepareStatement("delete from issuebooks where bookcallno=? and studentid=?");
+			ps.setString(1,bookcallno);
+			ps.setInt(2,studentid);
+			status=ps.executeUpdate();
+			}
+			
+			con.close();
+		}catch(Exception e){System.out.println(e);}
+		return status;
+	}
+	public static int updatebook(String bookcallno){
+		int status=0;
+		int quantity=0,issued=0;
+		try{
+			Connection con=DB.getConnection();

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *con*. The resource is referenced in statements at the following lines: 30, 39. The resource closure statement is at line: 46. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Connection.prepareStatement. Either a) close *con* in a try-finally block or b) close the resource by declaring *con* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).